src/library_dylink.js

```diff
@@ -256,60 +256,106 @@ var LibraryDylink = {
   // returns the side module metadata as an object
   // { memorySize, memoryAlign, tableSize, tableAlign, neededDynlibs}
   $getDylinkMetadata: function(binary) {
-    var next = 0;
+    var offset = 0;
+    var end = 0;
+
+    function getU8() {
+      return binary[offset++];
+    }
+
     function getLEB() {
       var ret = 0;
       var mul = 1;
       while (1) {
-        var byte = binary[next++];
+        var byte = binary[offset++];
         ret += ((byte & 0x7f) * mul);
         mul *= 0x80;
         if (!(byte & 0x80)) break;
       }
       return ret;
     }
 
+    function getString() {
+      var len = getLEB();
+      offset += len;
+      return UTF8ArrayToString(binary, offset - len, len);
+    }
+
+    var name = 'dylink.0';
     if (binary instanceof WebAssembly.Module) {
-      var dylinkSection = WebAssembly.Module.customSections(binary, "dylink");
+      var dylinkSection = WebAssembly.Module.customSections(binary, name);
+      if (dylinkSection.length === 0) {
```

Comment:
so when using the wasm JS API, checking the length of the returned object is the best way to tell whether a section exists in the module? that seems kind of ugly.

Reply:
It's confusing, but there can be more than one custom section with the same name. So that method returns a list of all the ones that match. And the list is empty if none do.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module/customSections#return_value

Reply:
Ah  I forgot about duplicate sections. that makes sense.